kong/tools/utils.lua

```diff
@@ -589,6 +589,12 @@ _M.get_response_type = get_response_type
 _M.get_error_template = get_error_template
 
 
+function _M.get_runtime_data_path(prefix)
+  local prefix = pl_path.normpath(prefix)
```

Comment:
I am wondering if `normpath` is enough here. There are several different values that could be passed in for `prefix` that are equivalent (dependent upon `$PWD`) but will obviously not hash to the same value:

* `./my-prefix`
* `my-prefix`
* `/path/to/my-prefix`
* `/path/to/foo/../my-prefix`
* `.`
* `/path/to/link` where `/path/to/link` is a symlink to `/path/to/my-prefix`
* `/path/to/link` where `/path/to/link` is a hard link to `/path/to/my-prefix`

[pl.normpath](https://lunarmodules.github.io/Penlight/libraries/pl.path.html#normpath) will resolve relative segments like `.` and `..`, but it does not normalize to an absolute path.

As for the symlink, I'm not 100% sure if we should catch that condition, but I'm leaning slightly towards "yes."

We obviously can't catch the hardlink case. Well, maybe we could if we used the inode as the source of our hash rather than a normalized path, but I dunno if that's a good solution.

Input-wise, there are couple different sources to think about:

1. At runtime: `kong.configuration.prefix` and `ngx.config.prefix()` are equivalent but not actually guaranteed to be `==`. Nginx does not normalize the prefix that it receives on the commandline, so it is technically possible for `kong.configuration.prefix` to be `/path/to/my-prefix` while `ngx.config.prefix()` is `./my-prefix`.
2. CLI: a user might call `kong reload -p ./my-prefix` or `kong reload -p /path/to/my-prefix`

Reply:
Whatever the end result is, I think `get_runtime_data_path()` would benefit greatly from some dedicated tests so we know where we stand with all input variants.

Reply:
good catch, there is also https://lunarmodules.github.io/Penlight/libraries/pl.path.html#abspath

Reply:
Yep, good catch! I'll use abspath and add some tests

Reply:
abspath seems to work for all of the cases except symlinks - do we want to make symlinks to directories hash to the same value for runtime data path? It's a bit of an edge case and I'm not sure if customers are likely to configure Kong this way.

Reply:
my 2cts; no. If you use the symlink, then keep using it. accessing the same path in different ways, then at your own risk.

If this leads to issues, then we can revisit or fix that edge case imho